packit/api.py

```diff
@@ -786,3 +787,54 @@ def clean(self):
         """ clean up stuff once all the work is done """
         # command handlers have nothing to clean
         logger.debug("PackitAPI.cleanup (there are no objects to clean)")
+
+    def validation(self) -> str:
+        """ Create output for PackageConfig validation."""
+        schema_errors = PackageConfigValidation(self.package_config).validate()
+
+        if not schema_errors:
+            return (
+                f"{self.package_config.config_file_path} is valid and ready to be used"
+            )
+
+        output = f"{self.package_config.config_file_path} does not pass validation:\n"
+        for field_name, errors in schema_errors.items():
+            output += self.validation_get_field_output(errors, field_name)
+        return output
+
+    def validation_get_field_output(
+        self,
+        errors: Union[list, dict],
+        field_name: str,
+        field_category: str = "* field",
+        level: int = 1,
+    ) -> str:
+        if isinstance(errors, list):
+            field_output = self.validation_format_field_output(
+                errors[0], field_name, field_category
+            )
+            return field_output
+
+        field_output = self.validation_get_field_item_output(errors, field_name, level)
+        return field_output
+
+    def validation_get_field_item_output(
+        self, errors: dict, field_name: str, level: int
+    ) -> str:
+        index_output = f"{level * '*'} field {field_name} has incorrect values:\n"
+        level += 1
+        for index, item_errors in errors.items():
+            index_type_error = self.validation_get_field_output(
+                item_errors,
+                index,
+                field_category=f"{level * '*'} value at index",
+                level=level,
+            )
+            index_output += index_type_error
+        return index_output
+
+    @staticmethod
+    def validation_format_field_output(
```

Comment:
this is an overkill to have a method for formatting a string, and it's only used in a single place - please just format string in-place above